### ISSUE TITLE
XML RPC addTorrent exception

### ISSUE THREAD

**Issue by amirotin** [tatsuhiro-t/aria2#480]
Trying to add torrent file via XML-RPC.
Got exception:
>11/16 16:04:41 [INFO] Executing RPC method aria2.addTorrent
>11/16 16:04:41 [INFO] Uploaded torrent data was saved as D:/Torrents/a652092b45ead8a1f8002d78f81d753775d45683.torrent
>11/16 16:04:41 [DEBUG] Exception caught
>Exception: [download_helper.cc:292] errorCode=25 Bencode decoding failed

Here is my code on python:

```python
baseurl = 'http://localhost:6800/rpc'
entry['location'] = 'D:\Torrent\some_file.torrent'
s = xmlrpclib.ServerProxy(baseurl)
r = s.aria2.addTorrent(config['token_str'], xmlrpclib.Binary(open(entry['location']).read()))
```

aria2 saves some part of torrent file (about 850 bytes) and rises exception.

**Comment by amirotin** [comment]
Well.. the problem is with `open(entry['location']).read()`. Python reads only part of file :(

**Comment by amirotin** [comment]
Please update documentation. In python example put `xmlrpclib.Binary(open('file.torrent', mode='rb').read())`

**Comment by tatsuhiro-t** [comment]
Thank you for notice this.  Fix committed via 9bce4eb
